swervelib/ctre/KrakenX60DriveControllerFactoryBuilder.java

```diff
@@ -0,0 +1,120 @@
+            TalonFXConfiguration motorConfiguration = new TalonFXConfiguration();
+
+            double sensorPositionCoefficient = Math.PI * moduleConfiguration.getWheelDiameter()
+                    * moduleConfiguration.getDriveReduction() / TICKS_PER_ROTATION;
+            double sensorVelocityCoefficient = sensorPositionCoefficient * 10.0;
+
+            if (hasVoltageCompensation()) {
+                motorConfiguration.voltageCompSaturation = nominalVoltage;
+            }
+
+            if (hasCurrentLimit()) {
+                motorConfiguration.supplyCurrLimit.currentLimit = currentLimit;
+                motorConfiguration.supplyCurrLimit.enable = true;
+            }
+
+            WPI_TalonFX motor = new WPI_TalonFX(id, canbus);
+            // CtreUtils.checkCtreError(motor.configAllSettings(motorConfiguration),
+            //         "Failed to configure Falcon 500"); // DROPPED FROM API
+
+            if (hasVoltageCompensation()) {
+                // Enable voltage compensation
+                motor.enableVoltageCompensation(true);
+            }
+
+            motor.setNeutralMode(NeutralMode.Brake);
+
+            motor.setInverted(moduleConfiguration.isDriveInverted() ? TalonFXInvertType.Clockwise
+                    : TalonFXInvertType.CounterClockwise);
+            motor.setSensorPhase(true);
+
+            // Reduce CAN status frame rates
+            // CtreUtils.checkCtreError(
+            //         motor.setStatusFramePeriod(StatusFrameEnhanced.Status_1_General,
+            //                 STATUS_FRAME_GENERAL_PERIOD_MS, CAN_TIMEOUT_MS),
+            //         "Failed to configure Falcon status frame period"); // DROPPED FROM API
+
+            return new ControllerImplementation(motor, sensorVelocityCoefficient);
+        }
+    }
+
+    private class ControllerImplementation implements DriveController {
+        private final WPI_TalonFX motor;
+        private final double sensorVelocityCoefficient;
+        private final double nominalVoltage = hasVoltageCompensation()
+                ? KrakenX60DriveControllerFactoryBuilder.this.nominalVoltage
+                : 12.0;
+
+        private ControllerImplementation(WPI_TalonFX motor, double sensorVelocityCoefficient) {
+            this.motor = motor;
+            this.sensorVelocityCoefficient = sensorVelocityCoefficient;
+        }
+
+        @Override
+        public Object getDriveMotor() {
+            return this.motor;
+        }
+
+        @Override
+        public void setReferenceVoltage(double voltage) {
+            motor.set(TalonFXControlMode.PercentOutput, voltage / nominalVoltage);
+        }
+
+        @Override
+        public double getStateVelocity() {
+            return motor.getSelectedSensorVelocity() * sensorVelocityCoefficient;
+        }
+
+        @Override
+        public double getDistanceMoved() {
+            throw new UnsupportedOperationException();
+            // return motor.getSelectedSensorPosition();
```

Comment:
if this is dead code, delete it. and is it ok that this is unsupported? it's not used anywhere? you're sure it doesn't have another way to get this? it would seem surprising if it doesn't